server/api/offline_batch_results.py

```diff
@@ -185,6 +185,89 @@ def record_offline_batch_results(
     return jsonify(status="ok")
 
 
+@api.route(
+    "/election/<election_id>/jurisdiction/<jurisdiction_id>/round/<round_id>/results/batch/",
+    methods=["POST"],
+)
+@restrict_access([UserType.JURISDICTION_ADMIN])
+def record_offline_batch_results_add(
+    election: Election,  # pylint: disable=unused-argument
+    jurisdiction: Jurisdiction,
+    round: Round,
+):
+    offline_batch_results = request.get_json()
+    validate_offline_batch_results(election, jurisdiction, round, offline_batch_results)
+
+    for batch_results in offline_batch_results:
+        for contest_choice_id, result in batch_results["choiceResults"].items():
+            db_session.add(
+                OfflineBatchResult(
+                    jurisdiction_id=jurisdiction.id,
+                    batch_name=batch_results["batchName"],
+                    batch_type=batch_results["batchType"],
+                    contest_choice_id=contest_choice_id,
+                    result=result,
+                )
+            )
+
+    db_session.commit()
+
+    return jsonify(status="ok")
+
+
+@api.route(
+    "/election/<election_id>/jurisdiction/<jurisdiction_id>/round/<round_id>/results/batch/<batch_name>",
+    methods=["PUT"],
+)
+@restrict_access([UserType.JURISDICTION_ADMIN])
+def record_offline_batch_results_update(
+    election: Election,  # pylint: disable=unused-argument
+    jurisdiction: Jurisdiction,
+    round: Round,
+    batch_name: str,
+):
```

Comment:
do we need to protect against two simultaneous edits to the same batch?

Reply:
No, if two desks update the same batch at the same time, I'm comfortable with last-write wins.